src/main/java/com/karankumar/bookproject/backend/statistics/PageStatistics.java

```diff
@@ -69,4 +123,23 @@ public Double calculateAveragePageLength() {
         return (booksWithPagesSpecified == 0) ? 0 :
                 Math.ceil(totalNumberOfPages / (float) booksWithPagesSpecified);
     }
+
+    /**
+     * @return the average page length for all books in the 'read' shelf in this year
+     * This average only includes books that have a page length specified
+     */
+    public Double calculateAveragePageLengthThisYear() {
+        booksWithPageCount.removeIf(book ->
+                (book.getDateStartedReading() == null ||
+                        book.getDateStartedReading().getYear() != Year.now().getValue()));
+        int totalNumberOfPages = booksWithPageCount.stream()
+                                                   .mapToInt(Book::getNumberOfPages)
+                                                   .sum();
+        int booksWithPagesSpecified = booksWithPageCount.size();
+        if (booksWithPagesSpecified == 0) {
+            return null;
+        }
```

Comment:
This isn't needed as you return 0 below if `booksWithPagesSpecified` is 0

Reply:
I wanted to ask you about this issue, should this statistic be absent in UI if there are no books with number of pages specified? If null is returned this statistic wont show in UI, if 0 is returned statistic will be shown in UI with average number of pages = 0. How should backend handle this situation?

Reply:
@MatusMaliar Sorry, I missed this

> I wanted to ask you about this issue, should this statistic be absent in UI if there are no books with number of pages specified?

Yes

> If null is returned this statistic wont show in UI, if 0 is returned statistic will be shown in UI with average number of pages = 0. How should backend handle this situation?

This should only be the concern of the frontend